dynamic-tools/zap-extensions/wso2-jira-issue-creator/wso2jiraissuecreator/ExtensionJiraIssueCreater.java

```diff
@@ -0,0 +1,268 @@
+     **/
+
+    private static boolean stringToBool(String s) {
+        if (s.equals("1"))
+            return true;
+        if (s.equals("0"))
+            return false;
+        throw new IllegalArgumentException(s + " is not a bool. Only 1 and 0 are.");
+    }
+
+    public void createJiraIssues(String jiraBaseURL, String jiraUserName, String jiraPassword, String projectKey,
+            String asssignee, String high, String medium, String low,String issueLabel, String filterIssuesByResourceType) {
+
+        String project_key = projectKey;
+        String issueList[];
+        JiraRestClient jira = new JiraRestClient();
+        int issueCount;
+        String issue;
+
+        try {
+
+            String auth = new String(Base64.encode(jiraUserName + ":" + jiraPassword));
+            String BASE_URL = jiraBaseURL;
+
+            XmlDomParser xmlParser = new XmlDomParser();
+            if (high.equals("1") || medium.equals("1") || low.equals("1")) {
+                issueList = xmlParser.parseXmlDoc(project_key, asssignee, stringToBool(high), stringToBool(medium),
+                        stringToBool(low), issueLabel,stringToBool(filterIssuesByResourceType));
+                issueCount = issueList.length; //get the issue count from the preset last index
+
+                if (issueCount != 0) { //proceed if the issue count is > 1
+                    for (int i = 0; i < issueCount; i++) { //create Issues in jira
+
+                        if (xmlParser.checkForIssueExistence(issueList[i],
+                                project_key)) { //update if the issue already exists
+
+                            xmlParser.updateExistingIssue(issueList[i], auth, BASE_URL, i);
+                            System.out.println("This is an existing issue");
+                        } else {                                             //create a new issue if not
+                            issue = jira.invokePostMethod(auth, BASE_URL + "/rest/api/2/issue", issueList[i]);
+                            System.out.println("Created Issue : " + issue);
+                        }
+                        System.out.println(i);
+                    }
+
+                } else { //abort if the issue count is = 0
+
+                    System.out.println("No issues found ");
+                }
+
+            } else {
+
+                System.out.println("No alert levels to create issues !!");
+            }
+
+        } catch (AuthenticationException e) { //authentication faliure
+            log.error(e.getMessage(), e);
+        }
+
+    }
+
+    public String[] loginUser() throws IOException, AuthenticationException {
+
+        Properties prop = new Properties();
+        InputStream input = new FileInputStream(Constant.getZapHome() + "/cred.properties");
+        prop.load(input);
+        String[] auth = new String[2];
+
+        if (!(prop.getProperty("jiraUrl").equals("")) && !(prop.getProperty("jiraUsername").equals("")) && !(prop
```

Comment:
use "".equals(prop.getProperty("jiraUrl") 

this way, if the property value is null, it avoids giving a null pointer exception.